src/max.ts

```diff
@@ -0,0 +1,66 @@
+import ReturnValueType from "./types/ReturnValueType";
+import { isAsyncIterable, isIterable } from "./_internal/utils";
+
+function sync(iterable: Iterable<number>) {
+  let n = NaN;
+  for (const a of iterable) {
+    if (Number.isNaN(a)) {
+      return a;
+    } else if (Number.isNaN(n)) {
+      n = a;
+    } else if (a > n) {
+      n = a;
+    }
```

Comment:
Since in most cases there will be no `NaN' value,
It looks a little better to write it like this:
```suggestion
    }
    if (a > n || Number.isNaN(n)) {
      n = a;
    }
```